packages/client/hmi-client/src/utils/stats.ts

```diff
@@ -11,3 +13,15 @@ export const stddev = (numberList: number[], usePopulation = false) => {
 			(numberList.length - (usePopulation ? 0 : 1))
 	);
 };
+
+// Get the mean absolute error between two maps.
+// Note this will only compare values that share the same key.
+export const mae = (map1: Map<number, number>, map2: Map<number, number>) => {
+	const sharedKey = [...map1.keys()].filter((key) => map2.has(key));
+	const error = _.meanBy(sharedKey, (key) => {
+		const firstValue = map1.get(key) as number;
+		const secondValue = map2.get(key) as number;
+		return Math.abs(firstValue - secondValue);
+	});
+	return error;
+};
```

Comment:
Maybe okay for now but slightly confusing, consider a more general usage scenario of a developing utility toolkit ...

People will generally have two or more time series that want to run the metric on, and not two maps. So from a design and API perspectives this is not particularly friendly - you are forcing people to transform their data before they can use this function, further more, you are transforming the Maps back into arrays for various operations here ... so maybe the question is why you chose to use Maps in the first place?

```
export const mae = (dataArray1: DataArray, dataArray2: DataArray) ..
```

Would be a better choice in my opinion, let me know if you want to have a little chat.

Reply:
Utilizing `export const mae = (dataArray1: DataArray, dataArray2: DataArray, keyField: string, valueField: string)`  as discussed
I am exporting the DataArray type rather than importing it from our simulation service as this is in a util file and should not depend on simulation service. Let me know if i should change this